packages/core/useMemoize/index.ts

```diff
@@ -0,0 +1,118 @@
+import { reactive } from 'vue-demi'
+
+type CacheKey = any
+
+/**
+ * Custom memoize cache handler
+ */
+export interface MemoizeCache<Key, Value> {
+  /**
+   * Get value for key
+   */
+  get (key: Key): Value | undefined
+  /**
+   * Set value for key
+   */
+  set (key: Key, value: Value): void
+  /**
+   * Return flag if key exists
+   */
+  has (key: Key): boolean
+  /**
+   * Delete value for key
+   */
+  delete (key: Key): void
+  /**
+   * Clear cache
+   */
+  clear (): void
+}
+
+/**
+ * Memoized function
+ */
+export interface MemoizedFn <Result, Args extends unknown[]> {
+  /**
+   * Get result from cache or call memoized function
+   */
+  (...args: Args): Result
+  /**
+   * Call memoized function and update cache
+   */
+  load (...args: Args): Result
+  /**
+   * Delete cache of given arguments
+   */
+  delete (...args: Args): void
+  /**
+   * Clear cache
+   */
+  clear (): void
+  /**
+   * Generate cache key for given arguments
+   */
+  generateKey (...args: Args): CacheKey
+  /**
+   * Cache container
+   */
+  cache: MemoizeCache<CacheKey, Result>
+}
+
+/**
+ * Reactive function result cache based on arguments
+ */
+export const useMemoize = <Result, Args extends unknown[]> (
+  resolver: (...args: Args) => Result,
+  options?: {
+    getKey?: (...args: Args) => CacheKey
+    cache?: MemoizeCache<CacheKey, Result>
+  },
+): MemoizedFn<Result, Args> => {
+  const cache = reactive(options?.cache || new Map<CacheKey, Result>())
```

Comment:
Vue 2 does not support making `Map` reactive. We would need either to get a around of not using it, or make this function as Vue 3 only

Reply:
I've added a fallback for Vue 2 to use a custom object instead of a `Map`. If you are happy with that, then I will squash the commits